### DISCUSSION TITLE
FTS on Android, iOS and Web

### DISCUSSION THREAD

**Discussion by jimmyff** [tekartik/sqflite#1146] [Q&A] [answered]
Hey, I've been building my app deploying native macos. I've been using `fts5` and the `bm25` ranking function. 

I've just got my android build working and discovered that `fts5` isn't included in android sqlite implementation only `fts4`. I've switched to that version but now I'm discovering that `bm25` isn't available.

My web version is now failing saying `fts4` not available although web was worked with `fts5` and `bm25` when I used the `package:sqlite`, perhaps web needs `fts5` or the wasm binary setup installed doesn't come with `fts`? 😵‍💫

Can anyone point me in the right direction to implementing a solution that will provide `fts` and `bm25` on Android, iOS and web?

Thanks very much!

**Comment by alextekartik** [comment, marked as answer]
Although I have not experimented FTS myself, I thik that one solution could be to use `sqflite_common_ffi` along with `flutter_sqlite_lib` which include the latest build of sqlite (including fts4) instead of the one on the platform that we don't control. It is just a matter a using a different factory: https://github.com/tekartik/sqflite/blob/master/sqflite_common_ffi/doc/using_ffi_instead_of_sqflite.md

I don't know about bm25 though.

**Reply by jimmyff** [reply]
Ahh perfect, thank you! I was using `sqflite_common_ffi` before switching to sqflite -I didn't realise it was compatible. `fts5` & `bm25` are both working again. Thanks
